src/base-apis.js

```diff
@@ -338,18 +338,29 @@ MatrixBaseApis.prototype.logout = function(callback) {
  * it is up to the caller to either reset or destroy the MatrixClient after
  * this method succeeds.
  * @param {object} auth Optional. Auth data to supply for User-Interactive auth.
- * @param {module:client.callback} callback Optional.
+ * @param {boolean} erase Optional. If set, send as `erase` attribute in the
+ * JSON request body, indicating whether the account should be erased. Defaults
+ * to false.
  * @return {module:client.Promise} Resolves: On success, the empty object
  */
-MatrixBaseApis.prototype.deactivateAccount = function(auth, callback) {
-    let body = {};
+MatrixBaseApis.prototype.deactivateAccount = function(auth, erase) {
+    if (typeof(erase) === 'function') {
+        throw new Error(
+            'deactivateAccount no longer accepts a callback parameter',
+        );
+    }
+
+    const body = { auth, erase };
     if (auth) {
-        body = {
-            auth: auth,
-        };
+        body.auth = auth;
+    }
+    if (erase !== undefined) {
+        body.erase = erase;
     }
+
     return this._http.authedRequestWithPrefix(
-        callback, "POST", '/account/deactivate', undefined, body, httpApi.PREFIX_UNSTABLE,
+        undefined, "POST", '/account/deactivate', undefined, body,
+        httpApi.PREFIX_UNSTABLE,
```

Comment:
can you fix this while you are here please

Reply:
fix what?

Reply:
PREFIX_UNSTABLE

Reply:
cf https://github.com/vector-im/riot-web/issues/5325

Reply:
I've moved it to R0, which I think is correct.